pilot/pkg/model/jwks_resolver.go

```diff
@@ -405,20 +422,30 @@ func (r *JwksResolver) refresher() {
 		case <-closeChan:
 			r.refreshTicker.Stop()
 			return
+		case jwtKeyData := <-jwksuriChannel:
+			hasErrors := true
+			for i := 0; i < 10; i++ {
```

Comment:
why are we looping 10x?

Reply:
Because it's a safe bet. Since this is in the background we are good with this.

Reply:
could you try to reuse the existing retry logic in `case <-r.refreshTicker.C:` and not duplicating it here?

Reply:
If I do that, I need to increase `networkFetchRetryCountOnRefreshFlow` or change the `refresh interval`. Do you think that would be ideal to do that ?